.github/workflows/pull-request-checks.yml

```
@@ -25,10 +25,14 @@ jobs:
         runs-on: ubuntu-latest
         steps:
             -   name: "If the event is to synchronize, we should remove the label from the PR"
-                if: ${{ github.event_name == 'pull_request_target' && github.event.action == 'synchronize' && github.event.pull_request.head.repo.full_name != github.repository && contains(github.event.pull_request.labels.*.name, env.CI_LABEL) }}
+                # also triggers on reopen to avoid a user clsoing, pushing changes and then reopening the PR
+                if: ${{ github.event_name == 'pull_request_target' && (github.event.action == 'synchronize' || github.event.action == 'reopened') && github.event.pull_request.head.repo.full_name != github.repository && contains(github.event.pull_request.labels.*.name, env.CI_LABEL) }}
```

Minor nit, I feel this reads a little cleaner; we're checking for a relevant value for action.
```suggestion
                if: ${{ github.event_name == 'pull_request_target' && contains(fromJSON('["synchronize", "reopened"]'), github.event.action) && github.event.pull_request.head.repo.full_name != github.repository && contains(github.event.pull_request.labels.*.name, env.CI_LABEL) }}
```